yapapi/executor/agreements_pool.py

```diff
@@ -104,7 +104,7 @@ async def _get_agreement(self) -> Optional[Agreement]:
             # in case there's more than one with this score.
             random.shuffle(offers)
             provider_id, offer = max(offers, key=lambda elem: elem[1].score)
-        except IndexError:  # empty pool
+        except ValueError:  # empty pool
```

Comment:
Using `max()` on empty list will raise `ValueError`, using `IndexError` is a leftover from an older version of the code.